src/gadgets/wnn.rs

```diff
@@ -331,11 +358,15 @@ impl<F: PrimeFieldBits> Circuit<F> for WnnCircuit<F> {
         config: Self::Config,
         mut layouter: impl Layouter<F>,
     ) -> Result<(), Error> {
-        let mut wnn_chip =
-            WnnChip::construct(config.wnn_chip_config, self.bloom_filter_arrays.clone());
+        let mut wnn_chip = WnnChip::construct(
+            config.wnn_chip_config,
+            self.bloom_filter_arrays.clone(),
+            self.binarization_thresholds.clone(),
+            self.input_permutation.clone(),
+        );
         wnn_chip.load(&mut layouter)?;
 
-        let result = wnn_chip.predict(layouter.namespace(|| "wnn"), self.inputs.clone())?;
+        let result = wnn_chip.predict(layouter.namespace(|| "wnn"), &self.image)?;
```

Comment:
Should the image be as a public input in `Advice`? As now we are proving that some image was converted into a valid representation